### ISSUE TITLE
Support for commands with multiple result sets

### ISSUE THREAD

**Issue by VitaliyMF** [nreco/data#35]
Lets add support for handling SQL commands that return multiple result sets:

- DbDataAdapter.Select(IDbCommand) overload (to allow custom command if needed)
- DbDataAdapter.SelectQuery() should handle multiple results sets (still not sure how to do that in best way).

**Comment by VitaliyMF** [comment]
I've added 2 new methods to `DbDataAdapter.SelectQuery` class:

- ExecuteReader
- ExecuteReaderAsync

With `DataReaderResult` class that appears as result of #37 now it is possible to read multiple result sets in simple but flexible way, example for stored procedure that returns multiple result sets:
```
(var companies, var contacts) = DbAdapter.Select("exec mystoredproc").ExecuteReader( (rdr) => {
	var companiesRes = new DataReaderResult(rdr).ToList<CompanyModel>();
	rdr.NextResult();
	var contactsRes = new DataReaderResult(rdr).ToList<ContactModel>();
	return (companiesRes, contactsRes);
});
```
Example for case if multiple result sets are returned as result of batch selects:
```
var batchCmdBuilder = new DbBatchCommandBuilder( dbFactory );
batchCmdBuilder.BeginBatch();
batchCmdBuilder.GetSelectCommand(new Query("companies"));
batchCmdBuilder.GetSelectCommand(new Query("contacts"));
var selectMultipleCmd = batchCmdBuilder.EndBatch();

(var companies, var contacts) = DbAdapter.Select(selectMultipleCmd).ExecuteReader( (rdr) => {
	var companiesRes = new DataReaderResult(rdr).ToList<CompanyModelAnnotated>();
	rdr.NextResult();
	var contactsRes = new DataReaderResult(rdr).ToList<ContactModel>();
	return (companiesRes, contactsRes);
});
```

**Comment by VitaliyMF** [comment]
Released in 1.0.2
